Updating branding to rtm

Anipik opened the pull request:
- git checkout release/5.0
- git merge release/5.0-rc2
https://github.com/dotnet/runtime/commit/d2e8221c4d18df2f01010eb3867f989fe9cfc7b3 causes merge conflicts
- resolve conflicts in favor of rc2
- update branding and push.

Still waiting for tactics to give a go ahead for this one.

jonfortescue committed:
test from windesktop

jonfortescue committed:
Cut the build out

jonfortescue committed:
no checkout bundle?

jonfortescue committed:
move to script

jonfortescue committed:
a smaller package

Anipik committed:
update branding (#41718)

MattGal committed:
Generate productVersion.txt and include in BlobArtifacts based off value of StabilizePackageVersion (#41721)

Anipik committed:
fix version (#41717)

jonfortescue committed:
remove some imports

jonfortescue committed:
move from build to targets

geoffkizer committed:
Remove System.Net.Connections and related features (#41648) (#41726)

* remove System.Net.Connections and related features

* cleanup csproj changes

* Update src/libraries/pkg/baseline/packageIndex.json

Co-authored-by: Cory Nelson <phrosty@gmail.com>

* Update src/libraries/System.Net.Sockets/src/System/Net/Sockets/Socket.Tasks.cs

Co-authored-by: Cory Nelson <phrosty@gmail.com>

Co-authored-by: Geoffrey Kizer <geoffrek@windows.microsoft.com>
Co-authored-by: Cory Nelson <phrosty@gmail.com>

Co-authored-by: Geoffrey Kizer <geoffrek@windows.microsoft.com>
Co-authored-by: Cory Nelson <phrosty@gmail.com>

carlossanlop committed:
Revert #40641

Reverts "Ensure FileStatus and FileSystemEntry Hidden and ReadOnly attributes are retrieved the same way"
(commit 6ed1e41e590c41862b993f99c8f401c1935a523a) which introduced a perf regression.

github-actions[bot] committed:
Fix cross-OS DAC metadata locator interface callback (#41762)

Co-authored-by: Mike McLaughlin <mikem@microsoft.com>

github-actions[bot] committed:
WASM: Add SHA* hash functions (#41761)

This ports the managed SHA* hash functions from https://github.com/microsoft/referencesource so we can use them on Browser/WASM where we don't have OpenSSL.

Enables the System.Security.Cryptography.Algorithms tests for these classes: `Tests run: 271, Errors: 0, Failures: 0, Skipped: 1. Time: 11.028473s`

Resolves https://github.com/dotnet/runtime/issues/40076

Co-authored-by: Alexander Köplinger <alex.koeplinger@outlook.com>

github-actions[bot] committed:
[release/5.0-rc2] Block casting to an interface marked ComInterfaceType.InterfaceIsIInspectable (#41702)

Co-authored-by: Jeremy Koritzinsky <jekoritz@microsoft.com>

github-actions[bot] committed:
Fix multiple version strings in coreclr shared libraries (#41735)

We were incorrectly adding the version.c source to all coreclr
libraries and not just the shared ones. That caused multiple
occurences of the version string in coreclr shared libraries
composed of sublibraries.
This change fixes it by adding that source file for shared
libraries only.
I've also discovered that libcoreclrtraceptprovider.so was
missing the version string completely, so I've added it there.

Co-authored-by: Jan Vorlicek <janvorli@microsoft.com>

github-actions[bot] committed:
Fix issue with covariant return method validation (#41734)

The ClassLoader::ValidateMethodsWithCovariantReturnTypes was being called
from a place where the relevant types were sometimes not fully loaded yet.
That resulted in an attempt to load the type being loaded recursively,
which was detected and the TypeLoadException was thrown.
This change fixes it by moving the call to the
ClassLoader::ValidateMethodsWithCovariantReturnTypes to the PushFinalLevels
when the level is greater than CLASS_LOAD_EXACTPARENTS, which is the
last stage of normal loading.

Co-authored-by: Jan Vorlicek <janvorli@microsoft.com>

github-actions[bot] committed:
[release/5.0-rc2] Strip binaries more aggressively (#41733)

* Strip binaries more aggressively

We are currently stripping only debug symbols, but there are also other
symbols that can be stripped without affecting debuggability or anything
else. We were stripping the binaries in corefx that way in the past,
but some time ago we have unified the stripping between the former
three repos and started to strip just the debugging symbols.

This change strips all unneeded symbols, which results in significant
reduction of the binaries sizes. For example, libcoreclr.so went down
from 8732272 to 7027024 or libSystem.Security.Cryptography.Native.OpenSsl.so
went down from 163376 to 120952.

* Fix FreeBSD

* Fix FreeBSD and OSX

We use nm command to extract g_dacTable symbol value from libcoreclr
for the DAC purposes. On OSX, there are no dynamic symbols, on
FreeBSD, the stripping strips all regular symbols and leaves
dynamic ones only. So the nm needs to get a -D argument for
non OSX.

Co-authored-by: Jan Vorlicek <janvorli@microsoft.com>

jonfortescue committed:
Merge branch 'release/5.0-rc2' into SigningInfoTest

jonfortescue committed:
Move WriteLinesToFile to correct location

jonfortescue committed:
PostBuildSign change

jonfortescue committed:
exclude productVersion.txt

jonfortescue committed:
remove item

carlossanlop committed:
Merge pull request #41820 from dotnet/backport/pr-41817-to-release/5.0-rc2

[release/5.0-rc2] Revert #40641

jonfortescue committed:
flat container

jonfortescue committed:
uhhh

jonfortescue committed:
publish flat container correctly

jonfortescue committed:
full build

NikolaMilosavljevic committed:
[release/5.0-rc2] Enable runtime closure verification (#41830)

* Enable closure validation

* Ensure dependency graph directory exists

* Netstandard should not be ignored

* Ensure Reports dir exists; VerifyClosure change moved to Arcade

* Merge

github-actions[bot] committed:
[release/5.0-rc2] Use dynamic port for BrowserDebugHost process (#41715)

Co-authored-by: Safia Abdalla <safia@microsoft.com>

jonfortescue committed:
quick adjustments on skipsigning

jonfortescue committed:
remove ItemsToSign if we're not postbuildsigning

jonfortescue committed:
a comma

github-actions[bot] committed:
[release/5.0-rc2] hold ref to temp keychain on OSX to avoid premature cleanup (#41882)

* hold ref to temp keychain on OSX to avoild premature cleanup

* feedback from review

Co-authored-by: wfurt <tweinfurt@yahoo.com>

github-actions[bot] committed:
Marking Matrix3x2, Matrix4x4, Plane, and Quaternion as Intrinsic (#41885)

Co-authored-by: Tanner Gooding <tagoo@outlook.com>

ViktorHofer committed:
Remove multi-targeting from runtime.depproj (#41666)

* Remove multi-targeting from runtime.depproj

and convert it to a csproj and use the NoBuild msbuild sdk. Simplify the
project logic in relation to IL vs native code as well.

github-actions[bot] committed:
[release/5.0-rc2] Update ICU (#41755)

Co-authored-by: EgorBo <egorbo@gmail.com>

github-actions[bot] committed:
[release/5.0-rc2] Bundle assemblies at 4K for linux arm64 (#41907)

Backport of #41809 to release/5.0-rc2

This fixes single-file deployments on Linux ARM64, which were segfaulting due to an incorrect computed address in R2R code. See #41832 for details about the issue.

github-actions[bot] committed:
Mark sizeValue as non faulting. (#41814)

The address of the ind can't be null because it is a prejited location.

However, the value of the indirection could be different, so should not mark as invariant.

Co-authored-by: Sergey Andreenko <seandree@microsoft.com>

github-actions[bot] committed:
Fix argument name in CompareInfo.IsPrefix (suffix -> prefix) (#41842)

Co-authored-by: Krzysztof Wicher <kwicher@microsoft.com>

github-actions[bot] committed:
Fix param name in ActivityContext..ctor (#41843)

Co-authored-by: Krzysztof Wicher <kwicher@microsoft.com>

github-actions[bot] committed:
[release/5.0-rc2] Properly handle byrefs in tailcall helper stubs (#41875)

* Properly handle byrefs in tailcall helper stubs

Switch to using ByReference instead of using stind.i/ldind.i and relying
on the JIT to report the locations being moved between.

Fixes #41555

* Move NextCallReturnAddress call back

Co-authored-by: Jakob Botsch Nielsen <Jakob.botsch.nielsen@gmail.com>

echesakov committed:
[release/5.0-rc2] Fix for folding of *(typ*)&lclVar for small types (#41838)

Fold *(typ*)&lclVar for small types into lclVar when:

1) it is *definitely load* and types of both indirection and local variable have the same signedness (e.g. bool and byte)
2) otherwise, fold the tree and mark the local node with GTF_VAR_FOLDED_IND
   and call fgDoNormalizeOnStore() on such nodes' parents in post-order morph.

github-actions[bot] committed:
Fix Crossgen2 package by copying the proper UCRT (#41890)

Co-authored-by: Anton Lapounov <antonl@microsoft.com>

github-actions[bot] committed:
[release/5.0-rc2] Restore exception compatibility in TcpListener.EndAccept*** (#41938)

Backport of #41745 to release/5.0-rc2

github-actions[bot] committed:
[release/5.0-rc2] Optimize the allocation and speed of ActivitySet/GetCustomProperty (#41908)

Co-authored-by: Tarek Mahmoud Sayed <tarekms@microsoft.com>

github-actions[bot] committed:
[release/5.0-rc2] SafeSocketHandle: avoid potential blocking of finalizer thread (#41747)

Backport of #41508 to release/5.0-rc2

* SafeSocketHandle: avoid potential blocking of finalizer thread

When the Socket is Disposed, it attempts to make all on-going operations
abort. It does this in a loop, and decides there are no on-going operations
when the reference count becomes zero and the handle gets released.

SafePipeHandle holds a reference to SafeSocketHandle. This prevents the
reference count to drop to zero, and causes the dispose to loop infinitly.

When the Socket is disposed from the finalizer thread, it is no longer used
for operations and we can skip the loop. This avoids blocking the finalizer
thread when the reference count can't drop to zero.

* When disposing from finalizer, fall back to ReleaseHandle

* Add test

* PR feedback

* Fix test

* PR feedback

* Refactor

* Refactor

* Log call to _handle.Dispose

* Handle null handle

Co-authored-by: Tom Deseyn <tom.deseyn@gmail.com>

epananth committed:
V3 publishing  (#41899)

github-actions[bot] committed:
Disable Http2_MultipleConnectionsEnabled_ConnectionLimitNotReached_ConcurrentRequestsSuccessfullyHandled (#41983)

Co-authored-by: Alexander Nikolaev <alnikola@microsoft.com>

dotnet-maestro[bot] committed:
Update dependencies from https://github.com/dotnet/arcade build 20200903.7 (#41896)

Microsoft.DotNet.GenFacades , Microsoft.DotNet.Build.Tasks.Feed , Microsoft.DotNet.Build.Tasks.Packaging , Microsoft.DotNet.Build.Tasks.SharedFramework.Sdk , Microsoft.DotNet.Build.Tasks.TargetFramework.Sdk , Microsoft.DotNet.CodeAnalysis , Microsoft.DotNet.GenAPI , Microsoft.DotNet.XUnitExtensions , Microsoft.DotNet.Helix.Sdk , Microsoft.DotNet.RemoteExecutor , Microsoft.DotNet.VersionTools.Tasks , Microsoft.DotNet.XUnitConsoleRunner , Microsoft.DotNet.Arcade.Sdk , Microsoft.DotNet.ApiCompat
 From Version 5.0.0-beta.20431.1 -> To Version 5.0.0-beta.20453.7

Co-authored-by: dotnet-maestro[bot] <dotnet-maestro[bot]@users.noreply.github.com>

github-actions[bot] committed:
Fix leaking ICorProfilerCallback10. (#41833)

Co-authored-by: Mikhail Pilin <mikhail@pilin.name>

github-actions[bot] committed:
[release/5.0-rc2] Force concurrent LdapConnection in new process (#41951)

* Ensure an initial non-concurrent call to OpenLDAP

OpenLDAP requires a single call for initialization before any other concurrent call.

This fixes asserts and segfaults we were seeing when calling OpenLDAP concurrently.

* Cleanup unused test changes

Co-authored-by: Eric StJohn <ericstj@microsoft.com>

mmitche committed:
Import configuration paths when building without PostBuildSign

mmitche committed:
Remove target for signing info

mmitche committed:
Merge remote-tracking branch 'upstream/release/5.0-rc2' into SigningInfoTest

github-actions[bot] committed:
Remove OSPlatformAttribute attributes from wasm linked output (#41806)

Fixes #40163

Co-authored-by: Marek Safar <marek.safar@gmail.com>

github-actions[bot] committed:
[release/5.0-rc2] Fix mono_wasm_get_obj_type performing a failed assembly load on every call (#41868)

Co-authored-by: Katelyn Gadd <kg@luminance.org>

ViktorHofer committed:
Update SDK min and target version to >= Preview8 (#41947)

* Update SDK min and target version to >= Preview8

mmitche committed:
Remove sign.props import

epananth committed:
Update arcade sdk (#42044)

* update arcade

* revert sdk

mmitche committed:
Merge remote-tracking branch 'upstream/release/5.0-rc2' into SigningInfoTest

github-actions[bot] committed:
[release/5.0-rc2] Fixes to nullable reference type annotations (#41845)

Co-authored-by: Krzysztof Wicher <kwicher@microsoft.com>
Co-authored-by: Buyaa Namnan <bunamnan@microsoft.com>

mmitche committed:
Revert "Remove sign.props import"

This reverts commit 6ef0a4969c4fd41ecebb81e655bbee1649760b20.

github-actions[bot] committed:
Update comments to match official Win32 API documentation. (#42088)

Co-authored-by: Aaron Robinson <arobins@microsoft.com>

github-actions[bot] committed:
[release/5.0-rc2] Implement SocketsHttpHandler.ConnectCallback (#42075)

* Implement SocketsHttpHandler.ConnectCallback

github-actions[bot] committed:
Temporarily disable arm64-intrin. UTF-16 validation (#42064)

Co-authored-by: Levi Broderick <levib@microsoft.com>

github-actions[bot] committed:
Remove machine state modifications on Unix (#42100)

Changing /proc/sys/kernel/core_pattern is handled by core-eng and should not be done by helix consuming repositories. Also removing a workaround that has been fixed since (__CrashDumpFolder=/cores) on macOS. Also removing ulimit which is handled by core-eng as well.

Co-authored-by: Viktor Hofer <viktor.hofer@microsoft.com>

dotnet-maestro[bot] committed:
Update dependencies from https://github.com/dotnet/arcade build 20200909.8 (#42080)

Microsoft.DotNet.Build.Tasks.Feed , Microsoft.DotNet.Build.Tasks.Packaging , Microsoft.DotNet.Build.Tasks.SharedFramework.Sdk , Microsoft.DotNet.Build.Tasks.TargetFramework.Sdk , Microsoft.DotNet.CodeAnalysis , Microsoft.DotNet.GenAPI , Microsoft.DotNet.GenFacades , Microsoft.DotNet.XUnitExtensions , Microsoft.DotNet.Arcade.Sdk , Microsoft.DotNet.Helix.Sdk , Microsoft.DotNet.RemoteExecutor , Microsoft.DotNet.VersionTools.Tasks , Microsoft.DotNet.XUnitConsoleRunner , Microsoft.DotNet.ApiCompat
 From Version 5.0.0-beta.20453.7 -> To Version 5.0.0-beta.20459.8

Co-authored-by: dotnet-maestro[bot] <dotnet-maestro[bot]@users.noreply.github.com>

mmitche committed:
Rework the implementation a bit

mmitche committed:
Merge remote-tracking branch 'upstream/release/5.0-rc2' into SigningInfoTest

mdh1418 committed:
[release/5.0-rc2] Backport Browser Unsupported Attribute Changes Properly (#42071)

Co-authored-by: Mitchell Hwang <mitchell.hwang@microsoft.com>
Co-authored-by: Maxim Lipnin <v-maxlip@microsoft.com>

mmitche committed:
Import Sign.props and repo configs conditionally

geoffkizer committed:
delete dead System.Net.Connections code (#42099)

Co-authored-by: Geoffrey Kizer <geoffrek@windows.microsoft.com>

AaronRobinsonMSFT committed:
Convert invalid C# uses of UnmanagedCallersOnly to IL. (#42146)

* Convert invalid C# uses of UnmanagedCallersOnly to IL for negative testing.

AaronRobinsonMSFT committed:
Disable UnmanagedCallersOnlyTest on WASM. (#42191)

github-actions[bot] committed:
[release/5.0-rc2] Redo annotation of Enumerable.Cast and friends to use oblivious TResult (#42216)

* Revert "Make Enumerable.Cast<T> and friends return IEnumerable<T?> (#40929)"

This reverts commit e75670889321611505115e4561a98e80732b58a0.

* Redo annotation of Enumerable.Cast and friends to use oblivious T

Some folks aren't happy with the fallout of Cast being annotated to return a `TResult?`, and others aren't happy with having it be `TResult`.  Consensus is to make the TResult oblivious, such that we don't declare one way or another what it is, since there's no good way to annotate it appropriately.

Co-authored-by: Stephen Toub <stoub@microsoft.com>

jkotas committed:
Unpin C# compiler version (#42174)

github-actions[bot] committed:
Fix ldtoken for generic RuntimeMethodHandle (#42141)

ldtoken for generic RuntimeMethodHandle in non-generic method triggered generic dictionary lookup due to normalization of MethodDescs for reflection. It lead to InvalidProgramException later.

Fixes #41674

Co-authored-by: Jan Kotas <jkotas@microsoft.com>

github-actions[bot] committed:
[release/5.0-rc2] Fix exception mapping during expect 100 cancellation. (#42014)

Co-authored-by: Marie Píchová <mapichov@microsoft.com>

layomia committed:
Add JsonNumberHandling.AllowReadingFromString as a JsonSerializer web default (#41539) (#42240)

* Add JsonNumberHandling.AllowReadingFromString as a JsonSerializer web default

* Add test assertion for number handling option in S.N.H.Json

* Test number-as-string behavior in System.Net.Http.Json

github-actions[bot] committed:
[release/5.0-rc2] Fix vtable chunk sharing with covariant returns (#42204)

* Fix vtable chunk sharing with covariant returns

There is an issue causing invalid sharing of a vtable chunk when a
method is overriden using method impl in a class and its parent class
and the slot in the parent class is in a previous vtable chunk. The
algorithm in MethodTableBuilder::ChangesImplementationOfVirtualSlot that
detects whether a slot in a vtable chunk was changed doesn't know that
the override will be propagated to all slots occupied by the method
being overriden. In the test attached to this fix, it leads to a stack
overflow due to an infinite recursion in the type system code. It could
also result in an overwrite of the method in the parent class, so when
another class derives from it, it would get a method from its sibling.

This change fixes it by checking if the parent method of a current slot
was overriden in some previous slot in the vtable.

* Gate slots scanning by presence of RequiresCovariantReturnTypeChecking

* Fix filtering condition in ChangesImplementationOfVirtualSlot

Co-authored-by: Jan Vorlicek <janvorli@microsoft.com>

github-actions[bot] committed:
[release/5.0-rc2] Fix covariant returns validation (#42203)

* Fix covariant returns validation

Recent fix to a problem with covariant returns validation has introduced
another problem related to code versioning. The validation method is now
called too late in the type load process which causes problem in
crossgening or in case code versioning is disabled.

This change fixes it by splitting the return type verification and
mutating changes like vtable slot updates into two separate phases.

* Fix missing precondition for calling SetRequiresCovariantReturnTypeChecking

Co-authored-by: Jan Vorlicek <janvorli@microsoft.com>

github-actions[bot] committed:
Fixing TargetHasAVXSupportCheck to read ECX not EDX (#42102)

Co-authored-by: Tanner Gooding <tagoo@outlook.com>

ViktorHofer committed:
Merge pull request #42269 from ViktorHofer/sdkupdate-RC2

[release/5.0-rc2] - Update SDK to final P8/RC1 and unpin compiler version

radical committed:
[release/5.0-rc2] Backport wasm debugger improvements, and fixes (#42057)

This fixes variable inspection and expression evaluation in the WASM debugger. It markedly improves developer experience in the debugger around watch expressions, locals, and object properties. It fixes the issues listed below, improves error handling and adds extensive tests.

Issues fixed:

#41818
#41744
#41846
mono/mono#20310
#41406
#40245
#41447
#41276
#41990

---

* [wasm][debugger]  Support deep member accesses for EvaluteOnCallFrame (#40836)

* [wasm][debugger] Support deep member accesses for EvaluteOnCallFrame

Eg. `obj.Property.X`, or `obj.Y + obj.Z.p`

Each of the member access expressions (like `a.b.c`) must be of only
primitive types. Though if the expression is a single member access (and
nothing else, like `"a.b.c"`), then that can be a non-primitive type.

This works by sending the expression to the browser, where it gets
resolved by `library_mono.js`. And that takes an easy route for doing
this, essentially just fetching the list of locals/properties, and using
that.

There are better ways to do this, that will be explored later.

* [wasm][debugger][tests] Remove some debug spew

* [wasm][debugger] fix formatting with dotnet-format

(cherry picked from commit 907f7da59b40c80941b02ac2a46650adf3f606bc)

* [wasm][debugger] Fix expression evaluation when it is a reference (#41135)

* [wasm][debugger] Fix expression evaluation when it is a reference

.. preceded by spaces. Eg: `"  foo.dateTime", or `"  foo.count"`

* Explanation of the fix:

- these particular expressions end up in the code path where we get a
SimpleMemberAccessExpression, and only one member access (like `a.b.c`)
was found.

- that code path had
    `return memberAccessValues[0]?["value"]?.Value<JObject>();`

  - which is incorrect, and we need to return `memberAccessValues[0]`,
  which is the value itself.

(cherry picked from commit ec59f65100d741539b00e308294cc53f04fc326b)

* [wasm][debugger] Correctly handle empty, and whitespace-only strings (#41424)

There are two cases being fixed here:

1. str=='', or str=' '
	- We check `str_value == 0`, and for the above cases JS returns
	true, due to type coercion.
	- So, we show the result as a null string.

2. str==null
	- debugger.c adds the value for this with `mono_wasm_add_typed_value ("string", NULL, 0)`
	- the second argument is converted to a string with
	  `Module.UTF8ToString(..)`, but when it's `0`/NULL, we get an
	  empty string. And that becomes a null string, because of (1).

Fixing this by using `===` operator to avoid type coercion.

Fixes https://github.com/dotnet/runtime/issues/41276

(cherry picked from commit 0795094ee0002133ae499d445fdb6770c0f5e0a5)

* [wasm] Disable an extraneous debug message (#41468)

Recent debugger test runs were showing lot of
`CWL: Failed to lookup sequence point` messages.

These are being shown for cases like:

`CWL: list_frames: Failed to lookup sequence point. method: runtime_invoke_direct_void, native_offset: 56`

This is a warning, and doesn't need to be emitted by default.

(cherry picked from commit c4841c5b62cb23b8d27fc7b675bb0d5515fbf827)

* [wasm][debugger] Breakpoint stopping after it's removed (#41479)

* Remove unnecessary WriteLine.
Fix #41447.

* Creating unit test about remove breakpoint.

* Implementing @radical suggestions!

(cherry picked from commit 513ade6f40cde36c2663f0fbd6c8b0947439ba28)

* [wasm][debugger] Add support for surfacing inherited members (#41480)

* [wasm][debugger][tests] Update to use `TDateTime`

- this ensures that we check the datetime, and some property getters on
it, wherever we have a datetime.

* [wasm][debugger][tests] Add labels to more checks

* [wasm][debugger] Add support for surfacing inherited members

- surface inherited fields, and properties
- we try to support `Runtime.getProperties`'s two arguments:
    - `ownProperties`, and `accessorsOnly`

    - `ownProperties`: for JS, this means return only the object's own
    members (not inherited ones)
    - `accessorsOnly`: for JS, this means return all the getters

Actual implementation:

- In practice, VSCode, and Chrome debugger seem to only send
`{ ownProperties: true, accessorsOnly: false }`,
and `{ ownProperties: false, accessorsOnly: true }`. The combination of
which means - that we don't return any inherited fields!

- But we want to show inherited fields too, so to get that behavior we
essentially *ignore* `ownProperties`. IOW,

    - `ownProperties`: we return all fields, and properties
    - `accessorsOnly`: we return only the getters, including the
    inherited ones

- Another thing to note is the case for auto-properties
    - these have a backing field
    - and we usually return the backing field's value, instead of
    returning a getter
    - To continue with that, auto-properties are *not* returned for
    `accessorsOnly`

- The code in `mini-wasm-debugger.c` does handle these two arguments,
but that is currently disabled by not passing the args to debugger.c at
all
    - Instead, we get the *full* list of members, and try to filter it
    in `library_mono.js`
    - which includes handling property overrides, or shadowing by new
    properties/fields in derived classes

* [wasm][debugger][tests] Fix simple warnings

* [wasm][debugger][tests] Fix warnings introduced in this PR

* [wasm][debugger][tests] Fix indentation

* [wasm][debugger] Correctly handle local structs in async methods

- When we have a struct local in an async instance method, it doesn't
get expanded, since we have a containerId (the async object), and we can
expand/access it later.

- When the IDE asks us to expand it with `{accessorPropertiesOnly: true}`:
    - we get the expanded json, but `_filter_automatic_properties` tries
    to return just the accessors, but that doesn't handle the expanded
    members of nested structs!
    - That is done in `extract_and_cache_value_types`, which is run *after*
    `_filter_automatic_properties`, but by that time we have already
    lost the expanded members!

    - So, `_get_vt_properties` fails with `Unknown valuetype id`,
    because it doesn't have anything to return at that point.

- This is being solved by ignoring the getProperties args in case of
expanding valuetypes.
    - that means that we can correctly extract, and cache the whole
    object.
    - And after that, we can return accessors/others, based on the args.

* [wasm][debugger] Fix warnings in debugger-test-app, and turn on warnAsError

* For some cases, debugger seems to give the actual method name instead of MoveNext for async methods

(cherry picked from commit b25b2bc6a3e64b70023d479a96692d590046627f)

* [wasm][debugger] Add support for Nullable<T>  (#41559)

* [wasm][debugger] Add support for Nullable<T>

Return the value, or null.

Fixes https://github.com/mono/mono/issues/20310

* Address review feedback - merge functions

* [wasm][debugger] run dotnet-format on the debugger test app

* [wasm][debugger] simplify function sig, based on usage

- addresses review feedback from @lewing

* [wasm][debugger] Simplify the function further, based on @lewing's

.. excellent suggestion!

(cherry picked from commit 66f4b4b70412da19f1785f57d7ae349febb0cd99)

* [wasm][debugger] Show actual data for boxed values (#41562)

* [wasm][debugger] Add support for Nullable<T>

Return the value, or null.

Fixes https://github.com/mono/mono/issues/20310

* Address review feedback - merge functions

* [wasm][debugger] run dotnet-format on the debugger test app

* [wasm][debugger] simplify function sig, based on usage

- addresses review feedback from @lewing

* [wasm][debugger] Simplify the function further, based on @lewing's

.. excellent suggestion!

* [wasm][debugger] Show actual data for boxed values

Eg. `object o = "foobar"`

This will show the string `"foobar"`, instead of an object, in the
debugger.

(cherry picked from commit 4fd87bc4ce9f90bcaf82de3fc5fed18486d54ea6)

* [wasm][debugger] Small improvements to fail gracefully (#41713)

* [wasm][debugger] Instead of failing completely, skip the problematic

.. property. Some times we might not get a `value`, or `name`, or it
might instead have a `get`. Handle those cases correctly when combining
the name/value/get objects.

This showed up in case of a `MulticastDelegate`, where we didn't have a
`value`, and ended up incorrectly combining the name/value objects, thus
returning incorrect locals.

* [wasm][debugger] Handle MulticastDelegate, and events

- Essentially we just surface these as a symbol showing the type name

* [wasm][debugger] Fail gracefully when vt could not be expanded

* [wasm][debugger] Handle invalid scope ids

scope<0, or scope too high

- This does get filtered at the proxy level, but app side should be able
to handle it too

* [wasm][debugger] Handle invalid/missing/failed value descriptions

- Eg. missing because of invalid param/local id, or value description
failed because of some as yet unsupported type

* [wasm][debugger] Fix frame indexing

- `collect_frames`, `list_frames` - both iterate over stack frames. They
skip some frames. This debug proxy assigns a simple index to each of the
received(filtered) frames.

    - so, if we had `[ frame0, (skipped frame1), frame2 ]`, the proxy will
    have `[ frame0(index:0), frame2(index:1) ]`

- `describe_variables_on_frame` also iterates over stack frames, and
tries to find a given frame by an index. And this index is what the
proxy had assigned.
    - because some frames were skipped, this function also needs to skip
    the *same* ones, so that it can find the intended frame.

    - Instead of trying to keep these in sync, here the indexing is
    changed to be the real index found as we iterate over the frames.
    - And the proxy just uses that.
    - So, we will have `[ frame0(index:0), frame2(index:2) ]`

This showed up in a trace in aspnetcore, which was called via
reflection. And that frame didn't get added to the list because it was
not `MONO_WRAPPER_NONE`, which caused the indexing to get out of sync.

Fixes: https://github.com/dotnet/runtime/issues/41818

* fix warning: remove unused var

* rebase on master, fix errors

* Make frame indices returned from debugger.c, 0-based

- Earlier this 1-based, and it was being adjusted in `MonoProxy`.
- Based on @lewing's suggestion, changing this to be 0-based in
debugger.c, itself, thus removing the need to "fixup" in `MonoProxy`.

* dotnet-format fixes

(cherry picked from commit 2e4e75bdf7a4d5168f79bdad1d3a3acff43f538c)

* Fix wasm sample after that was broken after this PR: https://github.com/dotnet/runtime/pull/40478 (#41277)

(cherry picked from commit f384168c9cca2d5a4f3d5e66b9d033fd3fe3e68b)

* [wasm][debugger] Avoid infinite loop when we have a boxed `new object` (#42059)

* [wasm][debugger] Avoid infinite loop when we have a boxed `new object`

Eg. `object o = new object(); object o1 = o;`
- Avoid infinitely looping for `o1`

* [wasm][debugger] Handle valuetypes boxed in classes that are not

.. type `object`.

Prompted by @lambdageek's comment - https://github.com/dotnet/runtime/pull/42059#issuecomment-690310274

(cherry picked from commit b58eba3fda4f36f73500ff6d5a55de75d33b86dd)

Co-authored-by: Thays Grazia <thaystg@gmail.com>

github-actions[bot] committed:
[release/5.0-rc2] Fix the globalization test. (#42226)

* Fix the globalization test.

The failure is because Windows regresion. The change here is to avoid having the test fail because of that. The regression is tracked by Windows team to fix it.

* Add a comment

Co-authored-by: Tarek Mahmoud Sayed <tarekms@microsoft.com>

mmitche committed:
Sign symbol packages in post-build

mmitche committed:
Uniquify the symbol package signatures. Sign mscoredaccore*.dll and mscordbi.dll even with post-build signing

mmitche committed:
Exclude->Remove

github-actions[bot] committed:
[release/5.0-rc2] AccessViolation when using ValueTypes as a Service (#42213)

* AccessViolation when using ValueTypes as a Service

When using ValueTypes as services in DependencyInjection, we are generating incorrect IL in a few cases:

- When the ValueType is a top level service
- When the ValueType is in an IEnumerable of services

We were double unboxing the ValueType, which was causing AccessViolations.

This was a regression caused by a previous change when ValueTypes were used in constructor parameters of services.

To fix this, I lifted the unboxing into the VisitConstructor and VisitIEnumerable methods instead of forcing VisitConstant to always do the unboxing. VisitConstant doesn't have enough context to know whether it should do the unboxing or not.

Fix #42037

* Add transient tests

Co-authored-by: Eric Erhardt <eric.erhardt@microsoft.com>

mdh1418 committed:
[release/5.0-rc2] Backport Additional Browser Unsupported Attribute Changes (#42267)

* Mark System.IO.Pipes Unsupported at assembly level (#41976)

Co-authored-by: Mitchell Hwang <mitchell.hwang@microsoft.com>

* Mark some System.Net.Http APIs as unsupported on browser (#42074)

* Mark some System.Net.HttpListener APIs as unsupported on browser (#42063)

* Mark some System.Net.HttpListener APIs as unsupported on browser

* FIx windows builds

* Fix windows builds

* Remove browser annotations from the properties with windows-specific setters.

* Mark only setter of ExtendedProtectionPolicy

* Remove another redundant annotation

* Mark System.Security.Cryptography.Algorithms as unsupported on Browser WASM (#42010)

* Mark System.Security.Cryptography.Algorithms as unsupported on Browser WASM

* Address the comment regarding IncrementalHash.CreateHMAC

* Group System.* usings

* Restore the blank lines between the meta-comment and the code for all the HMAC classes

* Fix style issues

* Mark MD5 class as unsupported on Browser; clean up the api exclusion list and proj file.

* Using

* Style fix

Co-authored-by: Mitchell Hwang <mitchell.hwang@microsoft.com>
Co-authored-by: Maxim Lipnin <v-maxlip@microsoft.com>

mmitche committed:
Do not pass None for symbol packages until arcade/6192 is fixed

mmitche committed:
Do not automatically skip signing with PostBuildSIgn=true

github-actions[bot] committed:
[release/5.0-rc2] [mono] Preserve GetterAdapterFrame if CreateGetterDelegate is used (#42120)

Fixes https://github.com/dotnet/runtime/issues/41811

Co-authored-by: Aleksey Kliger <alklig@microsoft.com>

mmitche committed:
Base items to sign on DownloadedArtifactFile

mmitche committed:
Conditionalize on prepare-artifacts and use different group

mmitche committed:
Simplify and fix

mmitche committed:
Allow empty sign list when doing post build signing

github-actions[bot] committed:
[release/5.0-rc2] Console.Unix: fix terminal settings during Console.KeyAvailable (#42381)

* Console.Unix: fix terminal settings during Console.KeyAvailable

Arguments passed to configure the terminal are in the wrong order.
This causes Enter keys to be returned as ConsoleKey.J instead of
ConsoleKey.Enter.

* Add test

Co-authored-by: Tom Deseyn <tom.deseyn@gmail.com>

github-actions[bot] committed:
Use portable PDBs even on .NETFramework (#42353)

Previously we were building full / windows PDBs for .NETFramework
because at the time .NETFramework couldn't read portable PDBs
so things like StackTrace wouldn't include file/line info.

That has been fixed as of 4.7.2 and we can switch to portable PDB
everywhere.

Symbol archiving will still convert to windows PDBs where necessary so
that we don't break any consumption scenarios for folks on older
frameworks.

This is important to get the latest symbols features only available in
the portable PDB format.

Co-authored-by: Eric StJohn <ericstj@microsoft.com>

github-actions[bot] committed:
[release/5.0-rc2] Fix createdump Windows triage dumps (#42394)

Fix createdump Windows triage dumps

Add ld (i.e. ld_2_23.so) interpreter program headers to Linux coredumps. This is important to the windbg linux support.

github-actions[bot] committed:
[release/5.0-rc2] Fix MacOS createdump testing failures (#42391)

Fix MacOS createdump testing failures

There are a few things that are causing MacOS dumps not to be complete and causing the diagnostics SOS tests to fail:

1) The threads were completely stopped with the per thread suspend. Switch from thread_suspend to task_suspend.
2) .Stopping filtering out SM_EMPTY share_mode type memory regions. There are read/write sections that are needed in the dump by the tools

github-actions[bot] committed:
[release/5.0-rc2] Make EventPipeProviderCallbackData own the filter data (#42368)


Co-authored-by: David Mason <davmason@microsoft.com>

layomia committed:
Fix continuation of JSON deserialization (#42158) (#42359)

* Repro #42070

* formatting

* namespace

* Fix

* never forget the header

* More tests
- Test continuation at every position inside the tested object
- Many member with primitive and nullable types
- One more level of nested object
- All combinations of class/struct for tested and nested object
- tested and nested object with parametrized ctor for some properties

* Addressed feedback

* Test with original repro data from #42070

* custom converter to ensure the padding is written in front of the tested object

* rename

* test data moved to Strings.resx

Co-authored-by: devsko <devsko@users.noreply.github.com>

Commit:
[release/5.0-rc2] Support sending events from runtime to debugging proxy (#42407)

* [wasm][debugger] Add support for raising events in the app, intended to

.. be received by the debug proxy.

* [wasm][debugger] `mono_wasm_raise_event` -> `mono_wasm_raise_debug_event`

Based on @lewing's suggestion

Co-authored-by: Ankit Jain <radical@gmail.com>

mmitche committed:
Fixup

mmitche committed:
One more time

github-actions[bot] committed:
[release/5.0-rc2] Fix Windows AccessViolationException in FileSystemWatcher when monitoring network share for changes (#42420)

* Fix FileSystemWatcher crash due to malformed data coming from Windows.

* Address PR suggestions.

* Bring back assert for numBytes.

Co-authored-by: carlossanlop <carlossanlop@users.noreply.github.com>

Commit:
[release/5.0-rc2] Support loading assemblies and symbols into debugger via callback (#42403)

* Add test setup for debugging lazy-loaded assemblies
* Add mono_wasm_add_lazy_load_files callback
* Rely on assembly_load callback to register PDBs
* Address feedback on formatting
* Fix spacing on method invocations
* Address second round of feedback
* Transport assembly data as base64 string
* Revert automated whitespace changes
* [wasm][debugger] Fix some issues with lazy loading assemblies
- One of the problems I found was that we were sending lot of duplicate
assemblies.
    - this is a problem because we are converting them to base64, then
    sending them over the wire, unnecessarily.
    - Also, the proxy wasn't correctly dealing with them, and ended up
    processing them, and then sending out duplicate `scriptParsed`
    events!
- So, we try to solve this at the `mono_wasm_asm_loaded` point:
    - issue: the `MONO.mono_wasm_runtime_is_ready` check wasn't enough
    - What we want to do is to send the events only for assemblies that
      were not in the bundle, list of which we can get in
      `MONO.loaded_files`.
    - but that has the filename
    - But we don't seem to be getting the filename from the MonoImage,
      so we use the assembly name as a workaround.

    - So, as a heuristic (and trying not too add too much new stuff this
    close to rc2!), we check if the assembly name begins with `System.`,
    or `Microsoft.`, and if it does then we look for `$assembly_name.dll`
    in the the loaded filenames.
        - If found, then we can skip these!
- The other point in the Proxy:
    - where we avoid adding duplicate assemblies if we already have one
    keyed by the assembly name.
* Fix spacing in function definition
* [wasm][debugger] Use the correct assembly name for the search
* Fixing and changes things related to Embedded Pdb.
* Update check for loaded assemblies and initialize resolver once
* Changing the order of the checks.
* Fix indentation.
* [wasm][debugger][tests] Improve tests a bit
.. which includes setting breakpoint using a url, before loading the
dynamic assembly. This essentially tests that the bp request gets resolved
automatically when the assembly gets loaded.
- And add the embedded pdb test for `DuplicateAssemblyLoadedEvent*`
tests

Co-authored-by: Ankit Jain <radical@gmail.com>
Co-authored-by: Thays <thaystg@gmail.com>

mmitche committed:
Fix artifact source globbing

github-actions[bot] committed:
Console.Unix: fix inverted TreatControlCAsInput (#42449)

The logical negation was unintentionally removed while refactoring.

Co-authored-by: Tom Deseyn <tom.deseyn@gmail.com>

github-actions[bot] committed:
[release/5.0-rc2] Remove implicit anchoring optimization from Regex (#42409)

* Remove implicit anchoring optimization from Regex

In .NET 5 we added a bunch of optimizations to Regex.  One of them was a transform that optimized for the case where the pattern begins with `.*`.  If it does, then we insert an implicit anchor at the beginning in order to avoid unnecessary backtracking.  Imagine the pattern `.*a` and the pattern `bcdefghijklmnopqrstuvwxyz`.  This is going to start matching at `b`, find the next newline, and then backtrack from there looking for the `a`; it won't find it and will backtrack all the way, failing the match at that position.  At that point it'll bump to the next position, starting at `c`, and do it all over.  It'll fail, backtrack all the way, and bump again, starting at `d`, and doing it all over.  Etc.  The optimization recognizes that since `.` will match anything other than newline, after it fails to match at the first position, we can just skip all subsequent positions until the next newline, as they're all going to fail.

However, the optimization failed to take into account that someone can explicitly start a match in the middle of the provided text.  In that case, the implicitly added anchor will fail the match in the actual "Go" matching logic.

There are safe ways to do this optimization, e.g. introducing a variant of these anchors that let FindFirstChar skip ahead but that aren't considered for Go's matching purposes, but we can look at employing those for .NET 6.  For now for .NET 5, this commit just deletes the faulty optimization and adds a few tests that were failing it.

* Address PR feedback

Co-authored-by: Stephen Toub <stoub@microsoft.com>

github-actions[bot] committed:
Avoid reusing Regex.Replace Match objects with MatchEvaluator (#42453)

One of the allocation-related optimizations we made for Regex in .NET 5 was for Regex.Replace and Regex.Split to not allocate unnecessary Match objects.  Previously, every match was producing a Match object which was then used to implement the API's semantics.  But in the case of an API like `Replace(string, string)`, those Match objects never make their way out to user code, so we can avoid creating a new Match object each time and instead only create one Match object for the whole operation and just reuse it over and over.

However, one of the overloads of Regex.Replace accepts a MatchEvaluator, a delegate that's handed the Match object.  The intent is that the callback fishes out from the Match whatever it needs and doesn't hold onto the Match object, that ownership isn't transferred... but it turns out some applications are indeed storing these Match objects and assuming ownership... and the reuse behavior breaks that.

This PR tweaks the caching behavior to be parameterized, only reusing the match object from callers where the Match object is never handed out to user code.

Co-authored-by: Stephen Toub <stoub@microsoft.com>

github-actions[bot] committed:
Fix several BeginXx nullable annotations (#42454)

AsyncCallback and object state parameters in APM BeginXx methods should always be nullable.  Several in annotated libraries were annotated incorrectly.

Co-authored-by: Stephen Toub <stoub@microsoft.com>

mmitche committed:
[release/5.0-rc2] Fix artifact source globbing for extract (#42487)

* Fix artifact source globbing for extract
There was a change the extract AzDO task that fixed a bug where it appeared to be matching some subpaths without globbing.

If you specified foo/x* it would apparently match subpaths of foo. Runtime was relying on this behavior, as the artifact target directory would implicitly get a subdirectory which was the artifact name, but not including that artifact name in the globbing pattern.

Fix this by including ** before the archive.

* Update download-artifact-step.yml

CoffeeFlux committed:
[release/5.0-rc2] Add test for wasm loader regression, fix loading from bundle (#42465)

* Add LoaderLinkTest

* Disable LoadFromAssemblyName_ValidTrustedPlatformAssembly on Browser

* Fix mono netcore bundle loading algorithm

mmitche committed:
Merge remote-tracking branch 'upstream/release/5.0-rc2' into SigningInfoTest

mmitche committed:
Import livebuilds.targets and Configuration.props based on context

github-actions[bot] committed:
[release/5.0-rc2] free existingTrust when verifying certificate name on macOS (#41989)

* free existingTrust when verifying certificate name

* add back NL

* free existingTrust on error path as well

Co-authored-by: wfurt <tweinfurt@yahoo.com>

github-actions[bot] committed:
[release/5.0-rc2] Memory Leak in Microsoft.Extensions.Caching.Memory when handling exceptions (#42494)

* Memory Leak in Microsoft.Extensions.Caching.Memory when handling exceptions

When an exception is thrown inside MemoryCache.GetOrCreate, we are leaking CacheEntry objects. This is because they are not being Disposed properly, and the async local CacheEntryStack is growing indefinitely.

The fix is to ensure the CacheEntry objects are disposed correctly. In order to do this, I set a flag to indicate whether the CacheEntry.Value has been set. If it hasn't, Disposing the CacheEntry won't add it to the cache.

Fix #42321

* Fix another memory leak when one cache depends on another cache. The inner cache's entries will reference the outer cache entries through the ScopeLease object.

Null'ing out the CacheEntry._scope field when it is disposed fixes this issue.

Co-authored-by: Eric Erhardt <eric.erhardt@microsoft.com>

Commit:
Add support for associating BrowserDebugHost with parent process (#42551)

dotnet-maestro[bot] committed:
[release/5.0-rc2] Update dependencies from dotnet/arcade (#42338)

* Update dependencies from https://github.com/dotnet/arcade build 20200915.7

Microsoft.DotNet.Build.Tasks.Feed , Microsoft.DotNet.Build.Tasks.Packaging , Microsoft.DotNet.Build.Tasks.SharedFramework.Sdk , Microsoft.DotNet.Build.Tasks.TargetFramework.Sdk , Microsoft.DotNet.CodeAnalysis , Microsoft.DotNet.GenAPI , Microsoft.DotNet.GenFacades , Microsoft.DotNet.XUnitExtensions , Microsoft.DotNet.Arcade.Sdk , Microsoft.DotNet.Helix.Sdk , Microsoft.DotNet.RemoteExecutor , Microsoft.DotNet.VersionTools.Tasks , Microsoft.DotNet.XUnitConsoleRunner , Microsoft.DotNet.ApiCompat
 From Version 5.0.0-beta.20459.8 -> To Version 5.0.0-beta.20465.7

* Update dependencies from https://github.com/dotnet/arcade build 20200917.6

Microsoft.DotNet.Build.Tasks.Feed , Microsoft.DotNet.Build.Tasks.Packaging , Microsoft.DotNet.Build.Tasks.SharedFramework.Sdk , Microsoft.DotNet.Build.Tasks.TargetFramework.Sdk , Microsoft.DotNet.CodeAnalysis , Microsoft.DotNet.GenAPI , Microsoft.DotNet.GenFacades , Microsoft.DotNet.XUnitExtensions , Microsoft.DotNet.Arcade.Sdk , Microsoft.DotNet.Helix.Sdk , Microsoft.DotNet.RemoteExecutor , Microsoft.DotNet.VersionTools.Tasks , Microsoft.DotNet.XUnitConsoleRunner , Microsoft.DotNet.ApiCompat
 From Version 5.0.0-beta.20459.8 -> To Version 5.0.0-beta.20467.6

Co-authored-by: dotnet-maestro[bot] <dotnet-maestro[bot]@users.noreply.github.com>

EgorBo committed:
[release/5.0-rc2] Update ICU dependency (fixes ja-jp datetime format) (#42525)

kjpou1 committed:
[release/5.0-rc2] [wasm][http] Add support for Blob URLs (#42522)

* [wasm][http] Add support for Blob URLs

* Modify the string to read `Only 'http' 'https' and 'blob' schemes are allowed.` for browser

* Add more tests

* Add blob Uri marshal test

* Address review comments to add comments within code about intentions.

* Multiple review comments.

- Split Fact into theory instead of specific test for blob.
- Split large test into multiple tests as well as mark it Theory where appropriate.

* Rename

* Add message to other validations as well

* Change name

* Update src/libraries/System.Net.Http/src/Resources/Strings.resx

Co-authored-by: Stephen Toub <stoub@microsoft.com>

* Create partial classes for HttpUtilities to replace the proliferating use of TARGETS_BROWSER in the sources.

* Define and call `HttpUtilities.InvalidUriMessage` to provide the invalid message to be thrown.

* Fix CI Build

* Rename modules to follow current standards

Co-authored-by: Stephen Toub <stoub@microsoft.com>

github-actions[bot] committed:
[release/5.0-rc2] Fix Getting default parent span Id with root Parent (#42558)

Co-authored-by: Tarek Mahmoud Sayed <tarekms@microsoft.com>

github-actions[bot] committed:
[release/5.0-rc2] HTTP/3 interop fixes (#42366)

* Fix SETTINGS frame creation.
* Only call tracing when it is enabled.
* Add descriptions for stream read states.
* Fix: SETTINGS frame parsing not discarding bytes from ArrayBuffer. Fix: ownership of ArrayBuffer could have resulted in double-free of arrays.
* Encode the alt-svc header after pseudo-headers ("All pseudo-header fields MUST appear in the header field section before regular header fields.") (#42545)

Co-authored-by: Cory Nelson <phrosty@gmail.com>

SamMonoRT commented on src/coreclr/tests/issues.targets:
```diff
@@ -2719,4 +2719,9 @@
         </ExcludeList>
         <!-- End interpreter issues -->
     </ItemGroup>
```
Just confirming, the original file is based off RC1/2 or off main ?  (In main this file issues.target, has been moved to https://github.com/dotnet/runtime/tree/master/src/tests )

Anipik replied:
its based off rc2

github-actions[bot] committed:
[release/5.0-rc2] [wasm][debugger] Fix stepping out from an async method (#42263)

* Fix assert when calling set_set_notification_for_wait_completion_flag

* Fix compilation error.

* Merging #42299 to backport together with #42227.

Co-authored-by: Thays Grazia <thaystg@gmail.com>

jonfortescue committed:
Merge pull request #41889 from jonfortescue/SigningInfoTest

[release/5.0-rc2] Add signing information to asset manifest

github-actions[bot] committed:
[release/5.0-rc2] Fix mapping of files from bundle to only map the necessary part (#42528)

Mapping the whole bundle every time leads to VM space starvation. Which mostly matters on 32bit.

Co-authored-by: vitek-karas <vitek.karas@microsoft.com>

github-actions[bot] committed:
[release/5.0-rc2] Fix the mechanism getting Constructed Culture (#42496)

Co-authored-by: Tarek Mahmoud Sayed <tarekms@microsoft.com>

github-actions[bot] committed:
Fix createdump on the alpine distro (#42586)

The stack unwinding in createdump was looping forever on the same IP (different SP) when it
was attempting to unwind a frame that was in the ld module (typically libld*.so or on alpine
ld-musl-x86_64.so). The ld module on Alpine doesn't having any unwind info (GNU_EH_FRAME empty)
causing the libunwind code to loop on the same IP.

This was caused by a recent change in PR #42159 to createdump that included the ld in the
coredump. Createdump wouldn't attempt to unwind IPs in the ld module because it wasn't added
to the module lookup list.

Including the ld module is important to the windbg linux support so it will have to be revisited
in 6.0 by changing the remote unwinder to handle this case of the Alpine ld module.

Co-authored-by: Mike McLaughlin <mikem@microsoft.com>

github-actions[bot] committed:
Always generate resx default values in ComponentModel.DataAnnotations (#42604)

Since many resource strings in ComponentModel.DataAnnotations contain messages that will be shown to end-users, we generate the "default value" string into the SR code. When a trimmed app strips the system resources, these strings will be used instead of the resource keys.

Fix #42257

Co-authored-by: Eric Erhardt <eric.erhardt@microsoft.com>

kouvel committed:
[5.0 RC2] Fix to allow entering cooperative GC mode when the thread is in a forbid-suspend-for-debugger region (#42630)

- Port of https://github.com/dotnet/runtime/pull/42587 to 5.0 RC2
- Followup to https://github.com/dotnet/runtime/pull/40060
- In short timing windows if a thread is placed in a pending-suspend-for-debugger state while in a forbid-suspend-for-debugger region, from the above PR it would not suspend for the debugger but it was missed that it can also get stuck in a perpetual spin-wait while entering cooperative GC mode. This causes the thread to not suspend for the debugger (VS times out after waiting) and the thread can only progress by unmarking it for debugger suspension.
- Fixed to break the spin-wait by checking whether the thread is in the forbid region

Fix for https://github.com/dotnet/runtime/issues/42375

github-actions[bot] committed:
[release/5.0-rc2] Revert "Make Console.ReadKey() distinguish between CR and LF inputs" (#42477)

* Revert "Make Console.ReadKey() distinguish between CR and LF inputs (#37491)"

This reverts commit a62bfcbd6e03396e85232a73aeef10df9df5fa4d.

* Account for terminals that have ICRNL disabled

* Add manual tests that validates the current \n ReadKey() behaviour on Unix

Co-authored-by: Eirik Tsarpalis <eirik.tsarpalis@gmail.com>

geoffkizer committed:
[release/5.0-rc2] Implement SocketsHttpHandler.PlaintextStreamFilter (#42563)

* add SocketsHttpHandler.PlaintextStreamFilter

github-actions[bot] committed:
[release/5.0-rc2] Ensure Enumerable methods are preserved by System.Linq.Querable (#42653)

* Ensure Enumerable methods are preserved by System.Linq.Querable

When using System.Linq.Queryable against an in-memory collection, and trimming the application, the underlying Enumerable methods need to be preserved by the linker.

Annotating each Queryable method with a DynamicDependency to the corresponding Enumerable method will fix this scenario.

Fix #42536

* Respond to PR feedback

Co-authored-by: Eric Erhardt <eric.erhardt@microsoft.com>

github-actions[bot] committed:
[release/5.0-rc2] Return null from ClaimsPrincipal.Current if no thread principal. (#42638)

* Return null from ClaimsPrincipal.Current if no thread principal.

In the .NET Framework, the AppDomain's default PrincipalPolicy is
UnauthenticatedPrincipal. When falling back to the thread's principal,
this would no throw because a GenericPrincipal was returned.
If the policy was changed to NoPrincipal, ClaimsPrincipal.Current would
throw constructing the ClaimsPrincipal since there is no principal.
However that behavior was opt-in.

In .NET Core, the default policy was changed to NoPrincipal, which would
mean unless a principal was explicitly set, Current would always throw.

This changes the behavior to return null instead of throw for
NoPrincipal. This is to continue supporting the existing behavior in
.NET Core 3 of returning null if there is no principal, but continue
to allow falling back to the thread's principal.

* Code review feedback.

Co-authored-by: Stephen Toub <stoub@microsoft.com>

* Fix typos

Co-authored-by: Kevin Jones <kevin@vcsjones.com>
Co-authored-by: Stephen Toub <stoub@microsoft.com>

github-actions[bot] committed:
[release/5.0-rc2] Update intellisense package version to RC2 (#42221)

* Update intellisense package version to RC2

* Update intellisense package.

Co-authored-by: Santiago Fernandez Madero <safern@microsoft.com>

EgorBo committed:
[release/5.0-rc2][wasm] Update ICU (fixes #42328) (#42579)

Co-authored-by: Santiago Fernandez Madero <safern@microsoft.com>

mmitche committed:
[release/5.0-rc2] Sign CrossOSDiag packages even with post-build signing (#42688)

* Sign CrossOsDiag packages even in Post-Build

* Sign base as well as symbols packages

* Install the plugin anyway

* Use VS2019 pools

Anipik commented:
Update:- git merged again today. (There were no conflicts today)

thaystg committed:
[release/5.0-rc2][wasm][debugger] Fix line number information for hidden locations

thaystg committed:
[release/5.0-rc2][wasm][debugger] Fixing step over in an async method (#42699)

MihaZupan committed:
Fix System.Net.Sockets telemetry (#42726)

Anipik commented:
@ericstj @safern @mmitche can u take another look before i merge this ? i merged the rc2 changes after the latest approval.

Anipik committed:
Merge remote-tracking branch 'upstream/release/5.0-rc2' into HEAD

Anipik committed:
update branding and delete an extra file from merge conflicts

Anipik commented:
```
C:\git\runtime>git diff upstream/release/5.0-rc2
diff --git a/eng/Versions.props b/eng/Versions.props
index 6870068f2fb..9ac70f317ac 100644
--- a/eng/Versions.props
+++ b/eng/Versions.props
@@ -6,8 +6,7 @@
     <MajorVersion>5</MajorVersion>
     <MinorVersion>0</MinorVersion>
     <PatchVersion>0</PatchVersion>
-    <PreReleaseVersionLabel>rc</PreReleaseVersionLabel>
-    <PreReleaseVersionIteration>2</PreReleaseVersionIteration>
+    <PreReleaseVersionLabel>rtm</PreReleaseVersionLabel>
     <!-- Set assembly version to align with major and minor version,
          as for the patches and revisions should be manually updated per assembly if it is serviced. -->
     <AssemblyVersion>$(MajorVersion).$(MinorVersion).0.0</AssemblyVersion>

C:\git\runtime>
```
